server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -656,7 +657,9 @@ private static Boolean resolveFailureStoreFromTemplate(String indexName, Metadat
             ComposableIndexTemplate composableIndexTemplate = metadata.templatesV2().get(template);
             if (composableIndexTemplate.getDataStreamTemplate() != null) {
                 // Check if the data stream has the failure store enabled
-                return composableIndexTemplate.getDataStreamTemplate().hasFailureStore();
+                return MetadataIndexTemplateService.resolveDataStreamOptions(composableIndexTemplate, metadata.componentTemplates())
+                    .map(DataStreamOptions.Template::toDataStreamOptions)
+                    .mapAndGet(DataStreamOptions::isFailureStoreEnabled);
```

Comment:
@gmarouli Am I right that there's a change to the semantic here? I think that, previously, if the index template had a `data_stream` block without a `failure_store` property, it would have return false (the `ComposableIndexTemplate.hasFailureStore()` method returns primitive boolean and defaults to false; now it will return null. It seems like this could change the counters incremented by `IngestService` (see the second call to `resolveFailureStore.apply()` in there).

Or is it not possible to create a template in that state?

Reply:
(For the record: If there is a behavioural change, I don't currently have a view on whether it's good or bad. I just wanted to make sure we'd thought it through. It's late in the day so I'm just dumping state before logging off...)

Reply:
It sounds from the other thread as though this actually can't happen, because `failure_store` is required on the data stream template, although good defensive coding implies that this should do the right thing if that ever changed.

Reply:
Fixed it and added a test to ensure we will not change the semantics again.